src/validator.cpp

```diff
@@ -125,7 +125,9 @@ struct Validator::ValidatorImpl
     * @param child Connected variable to start checking from
     * @param check_list The helper string returned containing the list(s) of cyclic variables
     */
-    bool cycleVariableFound(VariablePtr &parent, VariablePtr &child, std::deque<libcellml::VariablePtr> &check_list);
+    bool cycleVariableFound(VariablePtr &parent, VariablePtr &child, 
+                            std::deque<libcellml::VariablePtr> &check_list,
+                            std::vector<libcellml::VariablePtr> &all_variable_list);
```

Comment:
FWIW and if anything, `check_list` and `all_variable_list` ought to be renamed `checkList` and `allVariableList`, respectively. (This is the convention we use.)

Reply:
OK, thanks, will change that now

Reply:
Done :)